src/analysis/convex/function.lean

```diff
@@ -104,6 +104,29 @@ lemma strict_concave_on.subset {t : set E} (hf : strict_concave_on 𝕜 t f) (hs
   strict_concave_on 𝕜 s f :=
 ⟨hs, λ x y hx hy, hf.2 (hst hx) (hst hy)⟩
 
+section composition
+variables Q : Type*
```

Comment:
You should move this together with all the other `variables` declared at the beginning of the file.